source/include/time/cs_DayOfWeek.h

```diff
@@ -22,6 +22,15 @@ enum class DayOfWeek : uint8_t {
 
 typedef uint8_t DayOfWeekBitMask;
 
+inline uint8_t dayNumber(DayOfWeek day) {
+	for(auto i{0}; i < 7; i++) {
+		if (day == DayOfWeek(1<<i)) {
```

Comment:
Interesting implementation. If there's an underlying simple representation, I'd expect something like `DayOfWeek.getDayNumber()` which returns the internal `i`. Does it need to be an `enum class' derived from `uint8_t` or would have been a class with an internal `uint8_t` be just as good?

Reply:
There is an underlying representation, which is `(1<<i)` for what you call day `i`. This is so that you can easily do bitmasks of days with the same enum type. The tradeoff is that when you want a human readible day number you need to work a tiny bit. Hence the implementation. I anticipated that DayOfWeek would be used in switch statements at some point. Hence an enum.